src/asl.ts

```diff
@@ -4859,10 +4884,11 @@ export namespace ASL {
   export const compare = (
     left: ASLGraph.JsonPath,
     right: ASLGraph.Output,
-    operator: keyof typeof VALUE_COMPARISONS | "!="
+    operator: keyof typeof VALUE_COMPARISONS
```

Comment:
I found it confusing because the input was not handled by the function. Would rather functions be total over their input arguments. Since we normalize by inverting, i think it better to omit it from the types

Reply:
The compare function does support != (See the bottom)

Reply:
You're right.